src/MongoDB.Driver.Core/Core/Operations/RetryableWriteOperationExecutor.cs

```diff
@@ -86,7 +86,7 @@ public static TResult Execute<TResult>(IRetryableWriteOperation<TResult> operati
 
         public static async Task<TResult> ExecuteAsync<TResult>(IRetryableWriteOperation<TResult> operation, RetryableWriteContext context, CancellationToken cancellationToken)
         {
-            if (!context.RetryRequested || !AreRetryableWritesSupported(context.Channel.ConnectionDescription))
+            if (!context.RetryRequested || !AreRetryableWritesSupported(context.Channel.ConnectionDescription) || context.Binding.Session.IsInTransaction)
```

Comment:
The sync version had been updated for transactions but the async version was neglected.

I'm also changing the TransactionTestRunner to run the full transaction test suite using async APIs. That will be a separate PR.

I did test using a command line app that I could reproduce this issue before the fix and that the fix worked.